### PR TITLE
[ios, macos] Updated `maximumZoomLevel` description, cherry-picked #8818

### PR DESCRIPTION
**Pull request by jmkiley**
The `maximumZoomLevel` for a `MGLStyleLayer` is currently the zoom level that a layer is rendered up to, but not including.

Cherry-picked #8818 so that it can land in v3.5.3.

cc @boundsj  @captainbarbosa

### CODE REVIEW

File: platform/darwin/src/MGLStyleLayer.h [hunk 1]
```diff
@@ -63,7 +63,7 @@ MGL_EXPORT
 @property (nonatomic, assign, getter=isVisible) BOOL visible;
 
 /**
- The maximum zoom level at which the layer gets parsed and appears.
+ The maximum zoom level that the layer gets parsed and appears up to (but not including).
```
Comment (1ec5) [inline platform/darwin/src/MGLStyleLayer.h]: The multiple dangling prepositions are a bit awkward here. Also, I missed this before, but “parsing” a layer only makes sense in the context of style JSON, not runtime styling. How about:

> The exclusive upper bound of the range of zoom levels within which a map would display the layer.

For consistency, the `minimumZoomLevel` description would become:

> The inclusive lower bound of the range of zoom levels within which a map would display the layer.

File: platform/darwin/src/MGLStyleLayer.h [hunk 2]
```diff
@@ -63,12 +63,12 @@ MGL_EXPORT
 @property (nonatomic, assign, getter=isVisible) BOOL visible;
 
 /**
- The maximum zoom level at which the layer gets parsed and appears.
+ The exclusive upper bound of the range of zoom levels within which a map would display the layer.
```
Comment (friedbunny) [inline platform/darwin/src/MGLStyleLayer.h]: Hrm, the meaning of this is still a little hard to parse — I’m sure it’s technically correct (which is the best kind of correct), but could we offer an example or make it somehow simpler?

Comment (jfirebaugh) [inline platform/darwin/src/MGLStyleLayer.h]: I agree; the previous definition was easier to parse. I'm not convinced we actually need to make any documentation changes here. If anything we could simply remind the reader that this is a _floating point_ value, not an integer zoom level as they may be used to from raster-tile-based mapping libraries.
